enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,751 @@
+controller will either propagate the change or synchronize with the authoritative resource.
+
+The MachineSet sync controller will watch for the following condition:
+
+* `.spec.replicas` changes will be synced from the Cluster API MachineSet to the Machine
+  API MachineSet regardless of which is authoritative when the change originates from the
+  Cluster Autoscaler. As the Cluster Autoscaler will be configured to operate against Cluster
+  API resources only, there will be a need to identify when the Cluster Autoscaler has updated
+  a non-authoritative Cluster API resource so that the authoritative resource can be updated.
+  This will only occur when the sync controller observes and update to the replicas field from
+  the Cluster Autoscaler.
+
+#### MachineSet Validating Admission Policy Changes
+
+To ensure that the Cluster Autoscaler can update the `replicas` field of any Cluster API
+MachineSet, there will need to be a validating admission policy. This is required because
+the Cluster Autoscaler will be configured to only interact with Cluster API resources,
+regardless of authority. Furthermore, due to the natue of authoritative resources there will
+be restrictions around how users, and applications, can update non-authoritative resources.
+In cases where a Machine API MachineSet is authoritative, updates to the Cluster API mirror
+of that resource would be rejected.
+
+A validating admission policy to allow the Cluster Autoscaler service account to have write
+access to MachineSets will be created. The following examples show what the possible
+validating admission policies might look like:
+
+```
+matchConstraints:
+  resourceRules:
+  - apiGroups:   ["machine.openshift.io"]
+    apiVersions: ["v1beta1"]
+    operations:  ["UPDATE"]
+    resources:   ["MachineSet"]
+matchConditions:
+  # Only check requests coming from the cluster autoscaler service account.
+  - name: "check-only-cluster-autoscaler-service-account-requests"
+    expression: '(request.userInfo.username in [
+            "system:serviceaccount:openshift-machine-api:cluster-autoscaler",
+            ])'
+validations:
+  - expression: 'object.spec.replicas != oldObject.spec.replicas'
+    messageExpression: "Requested replica change is the same as current value"
+```
+
+```
+matchConstraints:
+  resourceRules:
+  - apiGroups:   ["cluster.x-k8s.io"]
+    apiVersions: ["v1beta1"]
+    operations:  ["UPDATE"]
+    resources:   ["MachineSet"]
+matchConditions:
+  # Only check requests coming from the cluster autoscaler service account.
+  - name: "check-only-cluster-autoscaler-service-account-requests"
+    expression: '(request.userInfo.username in [
+            "system:serviceaccount:openshift-machine-api:cluster-autoscaler",
+            ])'
+validations:
+  - expression: 'object.spec.replicas != oldObject.spec.replicas'
+    messageExpression: "Requested replica change is the same as current value"
+```
```

Comment:
Is this just a duplicate of the above block?

Reply:
mostly. the difference is targeting CAPI instead of MAPI.